saskatoon/harvest/forms.py

```diff
@@ -496,15 +541,52 @@ def clean_about(self):
             Harvest.Status.CANCELLED,
         ] and (not about or is_quill_html_empty(json.loads(about).get('html'))):
             raise forms.ValidationError(
-                _('Please fill in the public announcement to be published on the calendar.')
+                _("Please fill in the public announcement to be published on the calendar.")
             )
 
         return about
 
-    def clean(self):
+    def clean_equipment_point(self: Self) -> Optional[Organization]:
+        """Clear equipment point if no longer available or if harvest is not scheduled or ready"""
+        harvest_id = self.cleaned_data['id']
+        status = self.cleaned_data['status']
+        equipment_point = self.cleaned_data['equipment_point']
+        start = self.cleaned_data['start_date']
+        end = self.cleaned_data['end_date']
+
+        if equipment_point is None:
+            return equipment_point
+
+        if status not in [
+            Harvest.Status.SCHEDULED,
+            Harvest.Status.READY,
+            Harvest.Status.SUCCEEDED,
+        ]:
+            raise forms.ValidationError(
+                _(
+                    "You cannot reserve an equipment point if"
+                    " the harvest is not yet scheduled or ready."
+                )
+            )
+
+        harvest = Harvest.objects.get(pk=harvest_id) if harvest_id is not None else None
+
+        available_points = available_equipment_points(start, end, harvest)
+
+        if available_points.filter(pk=equipment_point.pk).count() != 1:
+            raise forms.ValidationError(
+                _("The " + equipment_point.civil_name + " equipment point is no longer available.")
```

Comment:
I think this is more straightforward for translations: 
```suggestion
                _("The {} equipment point is no longer available.").format(equipment_point.civil_name)
```